internal/k8s/client.go

```diff
@@ -48,28 +52,45 @@ type Client interface {
 	// Waits for the LoadBalancer to get a publicly available URL,
 	// then opens that URL in a web browser.
 	OpenService(ctx context.Context, lb LoadBalancer) error
+
+	// Opens a tunnel to the specified pod+port. Returns the tunnel's local port and a function that closes the tunnel
+	ForwardPort(ctx context.Context, namespace string, podID PodID, remotePort int) (localPort int, closer func(), err error)
 }
 
-type KubectlClient struct {
+type K8sClient struct {
 	env           Env
 	kubectlRunner kubectlRunner
+	restClient    k8sRestInterface
+	restConfig    *rest.Config
+	portForwarder PortForwarder
```

Comment:
what's the logic behind making this a func you attach to the struct rather than a method on the interface/struct?

Reply:
so that tests can use a fake PortForwarder